src/parser.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { model, Raml10Resolver } from "amf-client-js";
+import amf from "amf-client-js";
+import path from "path";
+import _ from "lodash";
+
+import { FatRamlResourceLoader } from "./exchange-connector";
+
+/**
+ * Parses a RAML file to an AMF model.
+ *
+ * @param filename The path to the RAML file to load
+ *
+ * @returns an AMF model of the RAML
+ */
+export async function parseRamlFile(
+  filename: string
+): Promise<amf.model.document.BaseUnit> {
+  const fileUri = `file://${path.resolve(filename)}`;
+
+  // We initialize AMF first
+  amf.plugins.document.WebApi.register();
+  amf.plugins.document.Vocabularies.register();
+  amf.plugins.features.AMFValidation.register();
+
+  await amf.Core.init();
+
+  const fatRamlResourceLoader = new FatRamlResourceLoader(
+    path.dirname(fileUri)
+  );
+  const ccEnvironment = amf.client.DefaultEnvironment.apply().addClientLoader(
+    fatRamlResourceLoader
+  );
+  const parser = new amf.Raml10Parser(ccEnvironment);
+
+  let model: model.document.BaseUnit;
+  try {
```

Comment:
I know this is copied from existing code but thought we can make changes while we are here -
Why is there a try-catch around this specific call?  Should we just throw the error since this is now a common function.
If not throwing the error then just change the error message to be more specific and use the logger instead of console.log

Reply:
done